sdk/keyvault/keyvault-keys/review/keyvault-keys.review.api.md

```diff
@@ -0,0 +1,295 @@
+    keyOps?: string[];
+    kid?: string;
+    kty?: JsonWebKeyType;
+    n?: Uint8Array;
+    p?: Uint8Array;
+    q?: Uint8Array;
+    qi?: Uint8Array;
+    t?: Uint8Array;
+    x?: Uint8Array;
+    y?: Uint8Array;
+}
+
+// @public
+export type JsonWebKeyCurveName = "P-256" | "P-384" | "P-521" | "P-256K";
+
+// @public
+export type JsonWebKeyEncryptionAlgorithm = "RSA-OAEP" | "RSA-OAEP-256" | "RSA1_5";
+
+// @public
+export type JsonWebKeyOperation = "encrypt" | "decrypt" | "sign" | "verify" | "wrapKey" | "unwrapKey";
+
+// @public
+export type JsonWebKeyType = "EC" | "EC-HSM" | "RSA" | "RSA-HSM" | "oct";
+
+// @public
+export interface Key {
+    keyMaterial?: JsonWebKey;
+    properties: KeyProperties;
+}
+
+// @public
+export interface KeyProperties extends ParsedKeyVaultEntityIdentifier {
+    readonly created?: Date;
+    enabled?: boolean;
+    expires?: Date;
+    id?: string;
+    notBefore?: Date;
+    readonly recoveryLevel?: DeletionRecoveryLevel;
+    tags?: {
+        [propertyName: string]: string;
+    };
+    readonly updated?: Date;
+}
+
+// @public
+export class KeysClient {
+    constructor(url: string, credential: TokenCredential, pipelineOrOptions?: ServiceClientOptions | NewPipelineOptions);
+    backupKey(name: string, options?: RequestOptions): Promise<Uint8Array | undefined>;
+    createEcKey(name: string, options?: CreateEcKeyOptions): Promise<Key>;
+    createKey(name: string, keyType: JsonWebKeyType, options?: CreateKeyOptions): Promise<Key>;
+    createRsaKey(name: string, options?: CreateRsaKeyOptions): Promise<Key>;
+    protected readonly credential: TokenCredential;
+    deleteKey(name: string, options?: RequestOptions): Promise<DeletedKey>;
+    static getDefaultPipeline(credential: TokenCredential, pipelineOptions?: NewPipelineOptions): ServiceClientOptions;
+    getDeletedKey(name: string, options?: RequestOptions): Promise<DeletedKey>;
+    getKey(name: string, options?: GetKeyOptions): Promise<Key>;
+    importKey(name: string, key: JsonWebKey, options?: ImportKeyOptions): Promise<Key>;
+    listDeletedKeys(options?: GetKeysOptions): PagedAsyncIterableIterator<KeyProperties, KeyProperties[]>;
+    listKeys(options?: GetKeysOptions): PagedAsyncIterableIterator<KeyProperties, KeyProperties[]>;
+    listKeyVersions(name: string, options?: GetKeysOptions): PagedAsyncIterableIterator<KeyProperties, KeyProperties[]>;
+    readonly pipeline: ServiceClientOptions;
+    purgeDeletedKey(name: string, options?: RequestOptions): Promise<void>;
+    recoverDeletedKey(name: string, options?: RequestOptions): Promise<Key>;
+    restoreKey(backup: Uint8Array, options?: RequestOptions): Promise<Key>;
+    updateKey(name: string, keyVersion: string, options?: UpdateKeyOptions): Promise<Key>;
+    readonly vaultBaseUrl: string;
+}
+
+// @public
+export type KeyWrapAlgorithm = "RSA-OAEP" | "RSA-OAEP-256" | "RSA1_5";
+
+// @public
+export interface NewPipelineOptions {
+    // (undocumented)
+    HTTPClient?: HttpClient;
```

Comment:
httpClient

Also, we should probably get this type into core-http to make sure we stay aligned on these option names. /cc @daviwil, maybe it already exists some place?

Reply:
At the moment it's just a shared pattern between the Storage and KeyVault data plane libraries.  I agree though, it'd be good to promote that concept to `core-http` so that other data plane libraries can build on it.

Reply:
This also came up as part of Cosmos review as they're adopting our pipeline, they will want to take these options as well /cc @southpolesteve. I'll file a bug on this.

Reply:
rename httpClient and should use shared types from  core-http